wagtail_wordpress_import/test/tests/test_import_hooks_items.py

```diff
@@ -343,125 +342,14 @@ def test_xml_items_cached(self):
-
-    def process_import(self, xml_stream):
-        self.importer = WordpressImporter(xml_stream)
-        self.logger = Logger(LOG_DIR)
-        self.importer.run(
-            logger=self.logger,
-            app_for_pages="example",
-            model_for_pages="TestPage",
-            parent_id=2,
-            page_types=["post"],
-            page_statuses=["publish"],
-        )
-        self.parent_page = Page.objects.get(id=2)
-        self.imported_pages = self.parent_page.get_children().all()
-
-    def test_get_hook_func_data(self):
-        fragment = """
-        <item>
-            <title>foo-item</title>
-            <link>https://www.example.com/foo-item/</link>
-            <pubDate>Tue, 13 Jul 2010 16:16:46 +0000</pubDate>
-            <guid isPermaLink="false">https://www.example.com/foo.jpg</guid>
-            <wp:post_id>100</wp:post_id>
-            <wp:post_date>2010-07-13 12:16:46</wp:post_date>
-            <wp:post_date_gmt>2010-07-13 16:16:46</wp:post_date_gmt>
-            <wp:post_modified>2010-07-13 12:16:46</wp:post_modified>
-            <wp:post_modified_gmt>2010-07-13 16:16:46</wp:post_modified_gmt>
-            <wp:post_name>foo-item</wp:post_name>
-            <wp:post_type>foo</wp:post_type>
-        </item>
-        <item>
-            <title>bar-item</title>
-            <link>https://www.example.com/foo-item/</link>
-            <pubDate>Tue, 13 Jul 2010 16:16:46 +0000</pubDate>
-            <guid isPermaLink="false">https://www.example.com/foo.jpg</guid>
-            <wp:post_id>100</wp:post_id>
-            <wp:post_date>2010-07-13 12:16:46</wp:post_date>
-            <wp:post_date_gmt>2010-07-13 16:16:46</wp:post_date_gmt>
-            <wp:post_modified>2010-07-13 12:16:46</wp:post_modified>
-            <wp:post_modified_gmt>2010-07-13 16:16:46</wp:post_modified_gmt>
-            <wp:post_name>bar-item</wp:post_name>
-            <wp:post_type>bar</wp:post_type>
-        </item>
-        <item>
-            <title>A title</title>
-            <link>https://www.example.com/a-title</link>
-            <description />
-            <content:encoded />
-            <excerpt:encoded />
-            <wp:post_id>44221</wp:post_id>
-            <wp:post_date>2015-05-21 15:00:31</wp:post_date>
-            <wp:post_date_gmt>2015-05-21 19:00:31</wp:post_date_gmt>
-            <wp:post_modified>2015-05-21 15:00:44</wp:post_modified>
-            <wp:post_modified_gmt>2015-05-21 19:00:44</wp:post_modified_gmt>
-            <wp:comment_status>open</wp:comment_status>
-            <wp:ping_status>closed</wp:ping_status>
-            <wp:post_name>a-title</wp:post_name>
-            <wp:status>publish</wp:status>
-            <wp:post_type>post</wp:post_type>
-        </item>
-        """
-        built_file = generate_temporay_file(
-            build_xml_stream(xml_items_fragment=fragment).read()
+        process_foo.assert_called_once_with(
+            mock.ANY,  # this is the queryset, and doesn't always match,
```

Comment:
Thanks Nick, after some investigation I found reconstructing the queryset still didn't work and that was because they are separate querysets in memory, thats the difference. Otherwise they are identical. The fix will be to use the same queryset from the Wordpress class. 

It means altering the Wordpress class to save the query to self.imported_pages and alter the following two import_string usages with self.imported_pages

Reply:
I've seen your change to this. It looks good.